src/NUnitFramework/framework/Attributes/RetryAttribute.cs

```diff
@@ -87,12 +87,15 @@ public RetryCommand(TestCommand innerCommand, int tryCount)
             /// <returns>A TestResult</returns>
             public override TestResult Execute(TestExecutionContext context)
             {
-                int count = _tryCount;
+                int currentAttempt = 0;
 
-                while (count-- > 0)
+                while (currentAttempt++ < _tryCount)
                 {
                     try
                     {
+                        // update CurrentAttempt property in test
+                        context.CurrentTest.Properties.Set(PropertyNames.CurrentAttempt, currentAttempt);
```

Comment:
I'm dubious about making the current retry count a property of the test. Test properties are persistent and pertain to the test itself. If results had properties (a change we have considered in the past) then it might make sense to include it there, since the retry count pertains to the __execution__ of the test rather than the test _per se_. Unfortunately, results don't have properties currently, although we have considered adding them in the past.

At the moment, based on the options available  to us, I think this is more a property of the execution context than of the test or the result.

Reply:
Thanks @CharliePoole, your comments and direction here are appreciated. Would you then suggest this be a public property of the _TestExecutionContext_ or that a _Properties_ dictionary be added to the _TestExecutionContext_ (I'm guessing the former as it is a smaller change, but the latter adds more flexibility for later additions).

Reply:
We have considered adding a dictionary to the `TestExecutionContext` - accessible through `TestContext` - for use by users. However, this count is really something internal and I think it's more in keeping with how we define other properties to make it a public property.

BTW, changes to `TestExecutionContext` are internal implementation changes in concept, even though the class is public for purpose of use by extensions. We will feel relatively free to change this in future where we would not feel quite so free to change what is exposed to general users via `TestContext`.

Reply:
reverted most of the changes here and am now incrementing a `CurrentRepeatCount` property in the _TestExecutionContext_ if the test is to be retried